app/utils.py

```diff
@@ -92,14 +92,33 @@ def json_only_has_required_keys(data, keys):
         abort(400, "Invalid JSON must only have {} keys".format(keys))
 
 
-def drop_foreign_fields(json_object, list_of_keys):
-    json_object = json_object.copy()
-    for key in list_of_keys:
-        json_object.pop(key, None)
-
+def drop_foreign_fields(json_object, list_of_keys, recurse=False):
```

Comment:
I was careful to make sure that the default was not to recurse, which should be the same behaviour as before. (And we do have test coverage there.)

I agree with 'no necessarily the same', but that's why the recursion should be optional, and ISTM that you need to know the structure in order to know whether and how to call this, regardless of how you implement the stripping at each level. In other words, whether the recursion is implemented manually (as currently) or automatically (per this PR), you have to know what the structure is in order to know whether stripping a given set of keys is a valid thing to do.

All that said, I think that the stripping-before-validation thing is a bit weird: ISTM it would be better to change the schema, to allow for optional links, and make this whole 'strip-out-the-links' thing just go away? Otherwise, we basically have two schema: one properly documented (but without links), and one that we actually use (that has links in it).